domain/model/state/controllerstate_test.go

```diff
@@ -1376,6 +1376,79 @@ func (s *stateSuite) TestGetModelUsersModelNotFound(c *gc.C) {
 	c.Assert(err, jc.ErrorIs, modelerrors.NotFound)
 }
 
+func (s *stateSuite) TestGetModelStateModelNotFound(c *gc.C) {
+	st := NewState(s.TxnRunnerFactory())
+	uuid := modeltesting.GenModelUUID(c)
+
+	_, err := st.GetModelState(context.Background(), uuid)
+	c.Check(err, jc.ErrorIs, modelerrors.NotFound)
+}
+
+// TestGetModelState is asserting the happy path of getting a model's state for
+// status. The model is in a normal state and so we are asserting the response
+// from the point of the model having nothing interesting to report.
+func (s *stateSuite) TestGetModelState(c *gc.C) {
+	st := NewState(s.TxnRunnerFactory())
+
+	mSt, err := st.GetModelState(context.Background(), s.uuid)
+	c.Check(err, jc.ErrorIsNil)
+	c.Check(mSt, gc.DeepEquals, model.ModelState{
+		Destroying:                   false,
+		Migrating:                    false,
+		HasInvalidCloudCredential:    false,
+		InvalidCloudCredentialReason: "",
+	})
+}
+
+// TestGetModelStateinvalidCredentials is here to assert  that when the model's
+// cloud credential is invalid, the model state is updated to indicate this with
+// the invalid reason.
+func (s *stateSuite) TestGetModelStateInvalidCredentials(c *gc.C) {
+	st := NewState(s.TxnRunnerFactory())
+	m, err := st.GetModel(context.Background(), s.uuid)
+	c.Assert(err, jc.ErrorIsNil)
+
+	credentialSt := credentialstate.NewState(s.TxnRunnerFactory())
+	err = credentialSt.InvalidateCloudCredential(
+		context.Background(),
+		m.Credential,
+		"test-invalid",
+	)
+	c.Assert(err, jc.ErrorIsNil)
+
+	mSt, err := st.GetModelState(context.Background(), s.uuid)
+	c.Check(err, jc.ErrorIsNil)
+	c.Check(mSt, gc.DeepEquals, model.ModelState{
+		Destroying:                   false,
+		Migrating:                    false,
+		HasInvalidCloudCredential:    true,
+		InvalidCloudCredentialReason: "test-invalid",
+	})
+}
+
```

Comment:
I know the migrating case has not been implemented.
Could we add an empty test with a TODO comment(or implement the test with Skip()) for migrating?

Reply:
ok